packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -52,119 +62,251 @@ class _DeepLinkPageState extends State<DeepLinkPage>
+                      showSpitScreen: showSpitScreen,
+                    ),
+                    _DataTableWithValidationDetails(
+                      tableView: TableView.singleUrlView,
+                      linkDatas: linkDatas,
+                      controller: controller,
+                      showSpitScreen: showSpitScreen,
+                    ),
+                  ],
+                ),
               ),
             ),
-          ],
-        ),
-        const SizedBox(height: denseSpacing),
-        ValueListenableBuilder<List<LinkData>>(
-          valueListenable: controller.linkDatasNotifier,
-          builder: (context, linkDatas, _) => _DataTable(
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _DataTableWithValidationDetails extends StatelessWidget {
+  const _DataTableWithValidationDetails({
+    required this.linkDatas,
+    required this.tableView,
+    required this.controller,
+    required this.showSpitScreen,
+  });
+  final List<LinkData> linkDatas;
+  final TableView tableView;
+  final DeepLinksController controller;
+  final bool showSpitScreen;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: _DataTable(
+            tableView: tableView,
             linkDatas: linkDatas,
-            bundleByDomain: controller.bundleByDomain,
+            controller: controller,
           ),
         ),
+        if (showSpitScreen)
+          Expanded(
+            child: ValueListenableBuilder<LinkData?>(
+              valueListenable: controller.selectedLink,
+              builder: (context, selectedLink, _) => _ValidationDetailScreen(
+                tableView: tableView,
+                linkData: selectedLink!,
+                controller: controller,
+              ),
+            ),
+          ),
       ],
     );
   }
 }
 
 class _DataTable extends StatelessWidget {
-  const _DataTable({required this.bundleByDomain, required this.linkDatas});
-  final bool bundleByDomain;
+  const _DataTable({
+    required this.linkDatas,
+    required this.tableView,
+    required this.controller,
+  });
   final List<LinkData> linkDatas;
+  final TableView tableView;
+  final DeepLinksController controller;
 
   @override
   Widget build(BuildContext context) {
-    final List<DataRow> rows = [
-      for (var i = 0; i < linkDatas.length; i++)
-        _buildRow(
-          context,
-          linkDatas[i],
-          color: MaterialStateProperty.all<Color>(
-            alternatingColorForIndex(i, Theme.of(context).colorScheme),
-          ),
-        ),
-    ];
+    final ColumnData<LinkData> domain = DomainColumn();
+    final ColumnData<LinkData> path = PathColumn();
 
-    return DataTable(
-      columns: const [
-        DataColumn(label: Text('OS')),
-        DataColumn(label: Text('Scheme')),
-        DataColumn(label: Text('Domain')),
-        DataColumn(label: Text('Path')),
+    return FlatTable(
```

Comment:
If you use `SearchableFlatTable` instead of `FlatTable` you should be able to get the search functionality you added above for free. You'll just have to make your `controller` mix in `SearchControllerMixin`

Reply:
I tried that and SearchableFlatTable highlights the search result, and I need to filter the result.

Reply:
I can use DevToolsClearableTextField, but I am still using `FlatTable` not `SearchableFlatTable` as this example is also using `FlatTable`